### ISSUE TITLE
Swap items between playlists

### ISSUE THREAD

**Issue by jaruba** [RSATom/WebChimera#78]
With a function like:

    plugin.swapItems(vlcPlayer.playlist.items[i],vlcPlayer2.playlist.items[j]);

I could make a Picture-in-Picture feature in WebChimera Player that would have continuas play (without needing to rebuffer) from one playlist item to another.

**Comment by RSATom** [comment]
Ok, I'll look. Thanks!

**Comment by RSATom** [comment]
Added possibility to swap vlcPlayers via `vlcPlayer.swap(vlcPlayer2)`. After that operation all content of vlcPlayers will be exchanged (attached video surfaces, playlist, current playing item and some other internal stuff).
Maybe some another functions will be useful, I'll think about it little more.

**Comment by jaruba** [comment]
That's great, then I can start thinking of a way to implement some sort of Picture in Picture feature with this. :)

**Comment by RSATom** [comment]
Is it will be enough for you to swap full playlist not only current item?

**Comment by jaruba** [comment]
Swapping items directly would be the best way to do it, but it should be possible with swapping playlists too.

Considering you mentioned that adding playlist items to the playlist does not connect to them until they are selected. I could clone the entire playlist, then play only the next item in the second playlist, when the current item in list 1 ends, then I swap it with the second playlist and remove the first one entirely.

If you think that swapping playlist items would not have an advantage for any other use, then I could try doing it like this to see if it works as expected.

**Comment by RSATom** [comment]
The reason why I've implemented only swap of full playlist - it was just simpler than swap only current items. But `swapCurrent` and `clonePlaylist` could be implemented too, the question is only in priority relative to other tasks.

**Comment by jaruba** [comment]
Then no, this is not a priority atm, as it can be done with ```.swap()``` too.

**Comment by RSATom** [comment]
Ok, then I will proceed with #86 and #83 and will start prepare new release...

**Comment by jaruba** [comment]
For me personally, I'm waiting on #84 and #88 the most.

#84 - If it works as expected, it would remove a big over head in the player and also repair a lot of bugs and limitations with subtitles. 

#88 - Limits me with 2 enhancements on Powder. As it is a pretty important feature to support drag and drop while playing too. (not sure if I understood correctly and this is even possible or not though)

**Comment by RSATom** [comment]
by #84 - did you mean adding extern subtitles support in plugin itself?

#88 - it's possible, but not easy - it needs write another patch to Qt. But I even don't sure how exactly It will be better fix it. So don't think I will do it for next release.

**Comment by jaruba** [comment]
**by #84 - did you mean adding extern subtitles support in plugin itself?**
yes, that's what I mean.

**Comment by RSATom** [comment]
then it will be better add it as separate issue - I'm afraid to forget about it. And yes, it's possible implement it before next release (and even it will not take much time)

**Comment by jaruba** [comment]
I'll add the issue then.

**Comment by RSATom** [comment]
now it's possible do following `vlcPlayer.playlist.add( vlcPlayer2.playlist.items[1] )` (https://github.com/RSATom/QmlVlc/commit/099f9bc4fef86e0ae45cf2e5c829ebbdf4a23f1f) so it seems we could close this issue?

**Comment by jaruba** [comment]
done
